mysticeti-core/src/net_sync.rs

```diff
@@ -178,6 +186,7 @@ impl<H: BlockHandler + 'static, C: CommitObserver + 'static> NetworkSyncer<H, C>
             epoch_close_signal,
             shutdown_grace_period,
             leader_timeout,
+            round_advanced_receiver,
```

Comment:
Tbh the plumbing is a bit more than I expected for this change. It seems to be due to having to pass a signal from syncer back to net sync. I thought net sync is responsible for networking only. Just curious, why net sync needs to be in the path of leader timeout? It would be nice if both round reaching quorum and leader timeout logic can exist along side other proposing logic.

Reply:
Yeah ideally I believe we would like to extract the timeout logic overall from the net sync, but for now I avoided triggering more refactoring as we might end up re-designing net sync anyways. This implementation is primarily a change to facilitate the main algo changes.